Muszilla/src/Controller/HomeController.php

```diff
@@ -0,0 +1,309 @@
+ * 
+ * @var object $user 
+ *    Variable to store the object of the User class.
+ * 
+ * @var object $upload_obj
+ *    Variable to store the object of the uploadCon class.
+ */ 
+class HomeController extends AbstractController {
+
+  private object $conn = '';
+  private object $music = '';
+  private object $connection_obj = '';
+  private object $upload_obj = '';
+  private object $user = '';
+
+  /**
+   * Construction to initialize the objects of the classes used in the 
+   * HomeController.
+   */
+  public function __construct() {
+    $this->music = new Music();
+    $this->connection_obj = new DbConnect();
+    $this->conn = $this->connection_obj->connect();
+    $this->upload_obj = new uploadCon();
+    $this->user = new User();
+  }
+
+  #[Route('/', name: 'app_login')]
+  /**
+   * Root Route to handle the user request to the login page.
+   *
+   * @param Request $request
+   *    Stores the credentials entered by the user.
+   * 
+   * @return Response
+   *    If the user credentiaals are valid, it redirect the user to the Home 
+   *    page,
+   *    otherwise redirects the user to the login page. 
+   */
+  public function login(Request $request): Response {
+    session_start();
+    if(isset($_SESSION['email'])) {
+      $user = $this->user->getUserData($_SESSION['email']);
+      return $this->render('Muszilla/home.html.twig', ['user'=>$user]);
+    }
+    if($_SERVER['REQUEST_METHOD'] == 'POST') {
+      try {
+        $data = $request->request->all();
+        $img = $this->upload_obj->uploadImage($data);
+        $register = $this->user->register($data, $img);
+      }
+      catch (Exception $e){
+        $error = $e;
+        $user = $this->user->getUserData($_SESSION['email']);
+        return $this->render('Muszilla/error.html.twig', ['error' => $error, 'user' => $user]);
+      }
+      
+    }
+    return $this->render('Muszilla/login.html.twig');
+  }
+
+
+  #[Route('/', name: 'app_root')]
+  /**
+   * Root route to handle the condition when the user tries to go back to the 
+   * login page after logging in.
+   *
+   * @return Response
+   *    Returns the home page if the user is already logged in, otherwise 
+   *    redirects the user to the home page.
+   */
+  public function root(): Response {
+    session_start();
+    if(isset($_SESSION['email'])) {
+      $user = $this->user->getUserData($_SESSION['email']);
+      return $this->render('Muszilla/home.html.twig', ['user'=>$user]);
+    }
+    return $this->render('Muszilla/login.html.twig');
+  }
+
+  #[Route('/logout', name: 'app_logout')] 
+  /**
+   * Function the destroy the user session and redirect to the login page if the
+   * user wants to log out.
+   *
+   * @return Response
+   *    If the user is logged in, then destroys the user session and redirects
+   *    to to the login page, otherwise if not logged in 
+   *    redirects to the login page.
+   */
+  public function logout(): Response {
```

Comment:
Why I am commenting about session?
Here assume if user not logged in and through url if some one try access this page then first session will start then verify user logged in or not if login then ok if not logged in then it will redirect to the login page but session is already start. Before user login what's the use of session varaible.

Reply:
Changes made as requested.